src/odemis/gui/util/widgets.py

```diff
@@ -334,13 +334,13 @@ def _update_progress(self):
         past = now - self._start
         left = max(0, self._end - now)
         prev_left = self._prev_left
+        ratio = past / (past + left)
```

Comment:
What happens if (past + left) = 0 and this code is ran?
It would create a ZeroDivisionError isn't it? Shouldn't that be prevented or catched?

Reply:
Indeed, good remark. This is the kind of code where it should never happen... but I'm sure we'll manage to hit it one day anyway! I've updated the code to handle gracely such case.